### PR TITLE
Fix `bundle init --gemspec` breaking when gemspec is invalid

### PR DESCRIPTION
**Pull request by colby-swandale**
Also added some specs to init what wasn't being tested.

### CODE REVIEW

**Comment by colby-swandale** [comment]
Seems like travis is having issues with the current  condition of Github. Will wait until morning and restart it if it still doesn't kick in then.

File: lib/bundler/cli/init.rb [hunk 1]
```diff
@@ -18,7 +18,13 @@ def run
           Bundler.ui.error "Gem specification #{gemspec} doesn't exist"
           exit 1
         end
+
         spec = Gem::Specification.load(gemspec)
```
Comment (segiddins) [inline lib/bundler/cli/init.rb]: Let's use the Bundler module method to load the gemspec?

Comment (segiddins) [inline lib/bundler/cli/init.rb]: `Bundler.load_gemspec_uncached`

Comment (colby-swandale) [inline lib/bundler/cli/init.rb]: Yes! 🎉

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**